File: Design Pattern/Easy/queue_using_two_stacks.cpp
// { Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
//User function Template for C++

// https://practice.geeksforgeeks.org/problems/Queue-using-stack/1#
// Implement a Queue using two stack s1 and s2.
class Queue {
    stack<int> input, output;
public:

    void enqueue(int x) {
        while(!input.empty())
        {
            output.push(input.top());
            input.pop();
        }
        
        input.push(x);
        while(!output.empty())
        {
            input.push(output.top());
            output.pop();
        }
        
    }

    int dequeue() {
        if(input.empty()) return -1;
        int res = input.top();
        input.pop();
        return res;
    }
};

// { Driver Code Starts.
int main()
{
    int T;
    cin>>T;
    while(T--)
    {
        Queue ob;

        int Q;
        cin>>Q;
        while(Q--){
        int QueryType=0;
        cin>>QueryType;
        if(QueryType==1)
        {
            int a;
            cin>>a;
            ob.enqueue(a);
        }
        else if(QueryType==2)
        {
            cout<<ob.dequeue()<<" ";

        }
        }
    cout<<endl;
    }
}
  // } Driver Code Ends

File: Design Pattern/Easy/queue_using_two_stacks.py
#User function Template for python3
# https://practice.geeksforgeeks.org/problems/Queue-using-stack/1#
# Implement a Queue using two stack s1 and s2.
class Queue:
    s1 = []
    s2 = []
    
    def enqueue(self,X):
        while(len(self.s1) != 0):
            self.s2.append(self.s1[-1])
            self.s1.pop()
            
        self.s1.append(X)
        
        while(len(self.s2) != 0):
            self.s1.append(self.s2[-1])
            self.s2.pop()
        
    def dequeue(self):
        if len(self.s1) == 0:
            return -1
            
        val = self.s1[-1]
        self.s1.pop()
        
        return val

#{ 
#  Driver Code Starts
if __name__ == '__main__':
    test_cases = int(input())
    for cases in range(test_cases) :
        ob=Queue()
        n = int(input())
        a = list(map(int,input().strip().split()))
        i = 0
        while i<len(a):
            if a[i] == 1:
                ob.enqueue(a[i+1])
                i+=1
            else:
                print(ob.dequeue(),end=" ")
            i+=1

        print()
# } Driver Code Ends